projects/simple/src/Example/Project.hs

```diff
@@ -1,19 +1,54 @@
-module Example.Project (topEntity, plus) where
+-- @createDomain@ below generates a warning about orphan instances, but we like
+-- our code to be warning-free.
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Example.Project where
 
 import Clash.Prelude
 
--- | Add two numbers. Example:
+-- Create a domain with the frequency of your input clock. For this example we used
+-- 50 MHz.
+createDomain vSystem{vName="Dom50", vPeriod=hzToPeriod 50e6}
+
+-- | 'topEntity' is Clash's equivalent of 'main' in other programming languages.
+-- Clash will look for it when compiling 'Example.Project' and translate it to
+-- HDL. While polymorphism can be used freely in Clash projects, a 'topEntity'
+-- must be monomorphic and must use non- recursive types. Or, to put it
+-- hand-wavily, a 'topEntity' must be translatable to a static number of wires.
 --
--- >>> plus 3 5
--- 8
-plus :: Signed 8 -> Signed 8 -> Signed 8
-plus a b = a + b
+-- Top entities must be monomorphic, meaning we have to specify all type variables.
+-- In this case, we are using the 'Dom50' domain, which we created with 'createDomain'
+-- and we are using 8-bit unsigned numbers.
+topEntity ::
+  Clock Dom50 ->
+  Reset Dom50 ->
+  Enable Dom50 ->
+  Signal Dom50 (Unsigned 8) ->
+  Signal Dom50 (Unsigned 8)
+topEntity = exposeClockResetEnable accum
+
+-- To specify the names of the ports of our top entity, we create a `Synthesize` annotation.
```

Comment:
```suggestion
-- To specify the names of the ports of our top entity, we create a @Synthesize@ annotation.
```

This isn't Haddock, but to suddenly switch to backticks for code segments gives me mental whiplash ;-)